numpy/core/tests/test_multiarray.py

```diff
@@ -3548,21 +3566,114 @@ def test_test_zero_rank(self):
 
 
 class TestPickling(object):
+    def test_highest_available_pickle_protocol(self):
+        try:
+            import pickle5
+        except ImportError:
+            pickle5 = None
+
+        if sys.version_info[:2] >= (3, 8) or pickle5 is not None:
+            assert pickle.HIGHEST_PROTOCOL >= 5
+        else:
+            assert pickle.HIGHEST_PROTOCOL < 5
+
+    @pytest.mark.skipif(pickle.HIGHEST_PROTOCOL >= 5,
+                        reason=('this tests the error messages when trying to'
+                                'protocol 5 although it is not available'))
+    def test_correct_protocol5_error_message(self):
+        array = np.arange(10)
+        f = io.BytesIO()
+
+        if sys.version_info[:2] in ((3, 6), (3, 7)):
+            # For the specific case of python3.6 and 3.7, raise a clear import
+            # error about the pickle5 backport when trying to use protocol 5
+            # without pickle 5
+            with pytest.raises(ImportError):
+                array.dump(f, protocol=5)
+
+            with pytest.raises(ImportError):
+                array.dumps(protocol=5)
+
+            with pytest.raises(ImportError):
+                array.__reduce_ex__(5)
+
+        elif sys.version_info[:2] < (3, 6):
+            # when calling __reduce_ex__ explicitly with protocol=5 on python
+            # version <3.6, instead of raising an AttributeError saying "pickle
+            # module has no attribute PickleBuffer", raise a ValueError
+            # saying that protocol 5 is not available for this python version
```

Comment:
Please remove 'instead of raising an AttributeError saying "pickle module has no attribute PickleBuffer"' from this comment. This is also a confusing implementation detail. The test and the comment should state and refer to what is expected in the user facing API / behavior, no what could happen if the implementation was wrong for one particular reason or another. There are too many ways to be wrong and it's not relevant.